bug(cloud-hooks): type-check broken on main — PLAN_RANK missing 'fleet' tier

## Problem
`apps/cloud-hooks/src/polar-notify.ts:19` declares `PLAN_RANK: Record<PlanId, number>` with only `free/pro/max/enterprise`, but `PlanId` gained `'fleet'` (B4 experiment tier, #2381) in `packages/pricing/src/plans.ts:35`.

`tsc --noEmit` in the cloud-hooks CI job fails:
```
src/polar-notify.ts(19,7): error TS2741: Property 'fleet' is missing in type '{ free: number; pro: number; max: number; enterprise: number; }' but required in type 'Record<PlanId, number>'
```

This is latent main breakage: it only surfaces when a PR touches cloud-hooks paths (`apps/cloud-hooks/**`, `packages/billing-webhook-core/**`, `packages/pricing/**`, or `cloudflare.yml`), which is how dependabot PR #2708 exposed it.

## Fix
Add `fleet` to `PLAN_RANK` ranked by monthly value: free(0) < pro($29) < max($99) < fleet($199) < enterprise(custom).

Found while triaging #2708 CI failures.

--- a/apps/cloud-hooks/src/polar-notify.ts
+++ b/apps/cloud-hooks/src/polar-notify.ts
@@ -15,12 +15,13 @@ import { BILLING_PLANS, monthlyEquivalentUsd, type PlanId } from '@chm/pricing'
 
 type Period = 'monthly' | 'yearly' | null
 
-/** Tier ranking for upgrade/downgrade detection. */
+/** Tier ranking for upgrade/downgrade detection, ordered by monthly value. */
 const PLAN_RANK: Record<PlanId, number> = {
   free: 0,
   pro: 1,
   max: 2,
-  enterprise: 3,
+  fleet: 3,
+  enterprise: 4,
 }
 
 export type TransitionCase =

